src/Frontend/Modules/Search/Ajax/Livesuggest.php

```diff
@@ -200,8 +202,11 @@ private function getRealData()
 
         // redirect if the request page doesn't exist
         if ($this->requestedPage > $this->pagination['num_pages'] || $this->requestedPage < 1) {
-            \SpoonHTTP::redirect(
-                FrontendNavigation::getURL(404)
+            throw new RedirectException(
```

Comment:
An ajax call that does a redirect. This seems rather strange to me? Shouldn't this just return a response with a 404 statuscode to the JavaScript code calling this method?